lib/matplotlib/_constrained_layout.py

```diff
@@ -175,12 +175,22 @@ def do_constrained_layout(fig, renderer, h_pad, w_pad,
                 _make_layout_margins(ax, renderer, h_pad, w_pad)
 
         # do layout for suptitle.
-        if fig._suptitle is not None and fig._suptitle._layoutbox is not None:
-            sup = fig._suptitle
-            bbox = invTransFig(sup.get_window_extent(renderer=renderer))
+        suptitle = fig._suptitle
+        print(suptitle)
+        do_suptitle = (suptitle is None or
+                       suptitle._layoutbox is None or
+                       not suptitle.get_in_layout())
+        do_suptitle = not do_suptitle
```

Comment:
That's a bit confusing.

There are three criteria that suptitle must meet to be layouted. That can be simply expressed by:

~~~
do_suptitle = (suptitle is not None
               and suptitle._layoutbox is not None
               and suptitle.get_in_layout())
~~~

Reply:
Ooops!  I'd meant to remove that...